tests/chart_tests/test_alertmanager.py

```diff
@@ -88,3 +92,39 @@ def test_alertmanager_customReceiver():
     )
     assert len(docs) == 1
     assert "sns-receiver" in docs[0]["data"]["alertmanager.yaml"]
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+def test_alertmanager_extra_volumes(kube_version):
+    test_extra_volumes_config = textwrap.dedent(
+        """
+        alertmanager:
+          extraVolumes:
+            - name: webhook-alert-secret
+              secret:
+                secretName: webhook-alert-secret
+          extraVolumeMounts:
+            - mountPath: "/var/webhook-alert-secret"
+              name: webhook-alert-secret
+              readOnly: true
+        """
+    )
+    values = yaml.safe_load(test_extra_volumes_config)
+    docs = render_chart(
+        kube_version=kube_version,
+        values=values,
+        show_only=["charts/alertmanager/templates/alertmanager-statefulset.yaml"],
+    )
+
+    assert "webhook-alert-secret" in [
+        x["name"] for x in jmespath.search("spec.template.spec.volumes", docs[0])
+    ]
+    assert "webhook-alert-secret" in [
+        x["name"]
+        for x in jmespath.search(
+            "spec.template.spec.containers[0].volumeMounts", docs[0]
+        )
+    ]
```

Comment:
We're actually trying to get away from using jmespath due to some syntactical sharp edges. 

```suggestion
    doc = render_chart(
        kube_version=kube_version,
        values=values,
        show_only=["charts/alertmanager/templates/alertmanager-statefulset.yaml"],
    )[0]

    c_by_name = get_containers_by_name(doc, include_init_containers=False)

    assert "webhook-alert-secret" in [
        x["name"] for x in doc["spec"]["template"]["spec"]["volumes"]
    ]
    assert "webhook-alert-secret" in [
        x["name"] for x in c_by_name["alertmanager"]["volumeMounts"]
    ]
```

Reply:
ACK - made the suggested changes, thanks!